geometry/r3_element_body.hpp

```diff
@@ -33,6 +39,10 @@ R3Element<Scalar>::R3Element(Scalar const& x,
                              Scalar const& y,
                              Scalar const& z) : x(x), y(y), z(z) {}
```

Comment:
In the body of the constructor:
```c++
static_assert(std::is_standard_layout<R3Element>::value, "blah");
```
This should ensure that the union is safe, because of the [rules pertaining to standard-layout unions](http://en.cppreference.com/w/cpp/language/data_members#Standard_layout) (or at least safer; `__m128d` is heavily implementation-defined, so it could in theory fail to be layout-compatible with doubles).

Reply:
Done.